### PR TITLE
Shimi stage

### PR DESCRIPTION
**Pull request by rutemartins**
Collective changes from @mlterryMitre and myself up to now. Merging with shimi so we can individually branch off a clean shimi branch.

### CODE REVIEW

File: spec/shr_oncology.txt [hunk 1]
```diff
@@ -55,24 +55,27 @@ Description:	"Diagnosis of cancer originating in the tissues of the breast, and
...
-// Perhaps make T, N, M separate observations
+// Perhaps make T, N, M separate observations (mlt_20180725: first pass WIP in this file.)
 // Do we want to follow Travis' approach of putting the Cancer Staging Encounter Phase in Encounter?
 Element:		BreastCancerStageTopic
 Based on:		ObservationTopic
 Description:	"The topic class for describing a breast cancer staging observation."
 				TopicCode is MTH#C2216702 "Malignant Neoplasm of Breast Staging"
 0..1			StageTimingPrefix				
-				EvaluationComponent			
+				EvaluationComponent
```
Comment (mlterryMitre) [inline spec/shr_oncology.txt]: zero-out EvaluationComponent

File: spec/shr_oncology.txt [hunk 2]
```diff
@@ -117,52 +384,55 @@ Description:	"The stage of a breast cancer. Different staging systems use differ
 				ObservationContext.ResultValue should be from http://loinc.org/vs/LL240-3
 
 
-EntryElement:	DCISNuclearGrade
-Concept:		MTH#C18513  // What is the proper LOINC code?
+EntryElement:	NuclearGrade
```
Comment (mlterryMitre) [inline spec/shr_oncology.txt]: After group discussion, revert references to NuclearGrade back to DCISNuclearGrade.

File: spec/shr_oncology.txt [hunk 3]
```diff
@@ -55,24 +55,27 @@ Description:	"Diagnosis of cancer originating in the tissues of the breast, and
...
 includes 0..1 	BreastCancerDistantMetastasesClassification
-				PanelMembers.Observation
 includes 0..1	HER2ReceptorStatus
 includes 0..1	EstrogenReceptorStatus
 includes 0..1	ProgesteroneReceptorStatus
-includes 0..1	DCISNuclearGrade
-includes 0..1	BreastCancerHistologicGrade	
+includes 0..1	BreastCancerNuclearGrade
+includes 0..1	BreastCancerHistologicGrade
+includes 0..1	Ki-67NuclearAntigen
```
Comment (mlterryMitre) [inline spec/shr_oncology.txt]: Move this out.  Not directly relevant to stage calculations.

File: spec/shr_oncology.txt [hunk 4]
```diff
@@ -55,24 +55,27 @@ Description:	"Diagnosis of cancer originating in the tissues of the breast, and
...
 includes 0..1 	BreastCancerPrimaryTumorClassification
 includes 0..1 	BreastCancerRegionalNodesClassification
 includes 0..1 	BreastCancerDistantMetastasesClassification
-				PanelMembers.Observation
 includes 0..1	HER2ReceptorStatus
 includes 0..1	EstrogenReceptorStatus
 includes 0..1	ProgesteroneReceptorStatus
-includes 0..1	DCISNuclearGrade
-includes 0..1	BreastCancerHistologicGrade	
+includes 0..1	BreastCancerNuclearGrade
```
Comment (mlterryMitre) [inline spec/shr_oncology.txt]: Move out BreastCancerNuclearGrade (DCISNuclearGrade) from BreastCancerStageTopic.

File: spec/shr_oncology.txt [hunk 5]
```diff
@@ -82,31 +85,295 @@ includes 0..1	BreastCancerHistologicGrade
...
+		Value:          CodeableConcept
+
 		
-		Element:		BreastCancerPrimaryTumorClassification
-		Concept:		SCT#78873005 "T category"
-		Based on:		CodedEvaluationComponent
-		Description:	"The size and extent of the primary tumor, based on criteria defined by the staging system being used."
-		Value:			CodeableConcept
-		0..1			StageSuffix
+		EntryElement:	TumorAssessment		// change to TumorAssessment
```
Comment (mlterryMitre) [inline spec/shr_oncology.txt]: TumorAssessment is not necessary now that the cardinality of PrimaryTumorClassification has 0..1.  Move TumorDimension and TumorMargin up one level.

File: spec/shr_oncology.txt [hunk 6]
```diff
@@ -82,31 +85,295 @@ includes 0..1	BreastCancerHistologicGrade
...
+					
+					Element:		TumorSecondaryDimensionSize
+					Description:	"The longest perpendicular diameter of the primary tumor."
+					Based on:		EvaluationComponent
+					Concept:		SCT#399738003 "Secondary tumor size (observable entity)"
+					Value:			Quantity with units UCUM#mm
+
+					Element:		SizeOfGrossTumorBed
+					Based on:		EvaluationComponent
+					Concept:		TBD			// mlt_to_rmb: needs new LOINC and SNOMED concept.
```
Comment (mlterryMitre) [inline spec/shr_oncology.txt]: @rutemartins to find appropriate code.  order of code selection:  LOINC (if observation), SNOMED, NCI Metathesaurus, local.

File: spec/shr_oncology.txt [hunk 7]
```diff
@@ -82,31 +85,295 @@ includes 0..1	BreastCancerHistologicGrade
...
+					// Do we need to record the method? whether the result of imaging, or study of a removed tumor after lumpectomy?	
+					0..0			LaboratoryObservationTopic.PanelMembers.Observation
+
+							Element:		TumorMarginDescription
+							Concept:		TBD
+							Based on:		CodedEvaluationComponent
+							Description:	"Description of the edge or border of tumor in situ by radiologist or of removed tumor by pathologist."
+							Value:			CodeableConcept from TumorMarginDescriptionVS
+
+							Element:		Cellularity
```
Comment (mlterryMitre) [inline spec/shr_oncology.txt]: Move back to comments section.

File: spec/shr_oncology.txt [hunk 8]
```diff
@@ -82,31 +85,295 @@ includes 0..1	BreastCancerHistologicGrade
...
+							Concept:		MTH#C4055283
+							Based on:		EvaluationComponent
+							Description:	"Percentage of cells in a sample that are cancerous"
+							Value:			Quantity with units UCUM#%
+
+							Element:		PercentageInSituCarcinoma
+							Based on:		EvaluationComponent
```
Comment (mlterryMitre) [inline spec/shr_oncology.txt]: Move back to comments section.

File: spec/shr_oncology.txt (hunk 3 above)
Comment (mlterryMitre) [reply spec/shr_oncology.txt]: completed change on local branch.  pending completion of group review to commit.

File: spec/shr_oncology.txt (hunk 5 above)
Comment (mlterryMitre) [reply spec/shr_oncology.txt]: completed change on local branch. pending completion of group review to commit.

File: spec/shr_oncology.txt [hunk 9]
```diff
@@ -82,31 +83,267 @@ includes 0..1	BreastCancerHistologicGrade
...
+	includes 0..1	TumorMargins
+
+		Element:   		StageSuffix  
+		Based on:       CodedEvaluationComponent
+		Concept:        TBD#TBD 
+		Description:    "A suffix used in conjuction with certain cancer stages, based on criteria defined by the staging system being used."
+		Value:          CodeableConcept
+
+	/*	
+		EntryElement:	TumorAssessment		// change to TumorAssessment
```
Comment (mlterryMitre) [inline spec/shr_oncology.txt]: Tumor Assessment now commented out and removed as the intermediate level.

File: spec/shr_oncology.txt [hunk 10]
```diff
@@ -82,31 +83,267 @@ includes 0..1	BreastCancerHistologicGrade
...
+					ObservationContext.ResultValue.CodeableConcept from TBD
+					LaboratoryObservationTopic.Focus value is type Identifier  // Lymph node identifier
+					LaboratoryObservationTopic.TopicCode is SCT#78873005 "T category"
+	0..0			LaboratoryObservationTopic.DiagnosticService  
+	0..0			LaboratoryObservationTopic.Specimen // also removed b/c this is determined lower
+	0..0			LaboratoryObservationTopic.Precondition
+	0..0			LaboratoryObservationTopic.FindingMethod
+			 		LaboratoryObservationTopic.EvaluationComponent
+	includes 0..1	StageSuffix
+					LaboratoryObservationTopic.PanelMembers.Observation
```
Comment (mlterryMitre) [inline spec/shr_oncology.txt]: @markkramerus - TumorAssessment now removed from BreastCancerPrimaryTumorClassification as the intermediate level.

File: spec/shr_oncology.txt (hunk 4 above)
Comment (rutemartins) [reply spec/shr_oncology.txt]: Leaving as is for now. See issue #115.

File: spec/shr_oncology.txt (hunk 6 above)
Comment (rutemartins) [reply spec/shr_oncology.txt]: @mlterryMitre to revisit whether `SizeOfGrossTumorBed` should be included for the purposes of staging. No changes at this point.

**Comment by mlterryMitre** [comment]
Merging noting that while it is still a work in progress, doing this will establish a clean work tree for further changes off of shimi.

**Pull request merged**